chalice/local.py

```diff
@@ -187,24 +187,43 @@ def do_OPTIONS(self):
         else:
             # Otherwise this is a preflight request which we automatically
             # generate.
-            self._send_autogen_options_response()
+            self._send_autogen_options_response(lambda_event)
 
     def _has_user_defined_options_method(self, lambda_event):
         # type: (EventType) -> bool
         route_key = lambda_event['requestContext']['resourcePath']
         return 'OPTIONS' in self.app_object.routes[route_key]
 
-    def _send_autogen_options_response(self):
-        # type:() -> None
+    def _send_autogen_options_response(self, lambda_event):
+        # type:(EventType) -> None
+        route_key = lambda_event['requestContext']['resourcePath']
+        route_dict = self.app_object.routes[route_key]
+        route_methods = list(route_dict.keys())
+
+        # Chalice ensures that routes with multiple views have the same
+        # CORS configuration, so if any view has a CORS Config we can use
+        # that config since they will all be the same.
+        cors_config = route_dict[route_methods[0]].cors
+        cors_headers = cors_config.get_access_control_headers()
+
+        # The keys will be ordered in python 3 but in python 2 they will not
+        # be. To assert this header is sent correctly we need to introduce
```

Comment:
Talked offline about this. The reason it is being done is that there is not an easy way to parse the raw body and pull out the specific header/value without using something like the `email` library. Given that this is being ran in `chalice local` it is fine that it is getting sorted as `chalice local` is just a test mode. If this was being done in the actual lambda function, it would be something we would not want to do and should add the extra logic in the tests.